src/main/java/ro/msg/learning/shop/model/OrderDetailId.java

```diff
@@ -0,0 +1,23 @@
+package ro.msg.learning.shop.model;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Embeddable;
+import java.io.Serializable;
+
+@Embeddable
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode
+public class OrderDetailId implements Serializable {

```

Comment:
You can group your composite ids into a sub package to be clear that they are not models/entities.